File: 1046/1046.lua
local valores = {}

for numero in string.gmatch(io.read(), "[^%s]+") do
    table.insert(valores, tonumber(numero))
end

local inicio = valores [1]
local fim = valores [2]

local total = 0

if inicio == fim then
    total = 24
elseif inicio > fim then
    fim = fim + 24
    total = fim - inicio
else
    total = fim - inicio
end

print(string.format("O JOGO DUROU %d HORA(S)", total))

File: 1046/1046.java
import java.io.IOException;
import java.util.Scanner;
 
public class Main {
 
    public static void main(String[] args) throws IOException {
 
        Scanner sc = new Scanner(System.in);
        
        int inicio = sc.nextInt();
        int fim = sc.nextInt();
        int total = 0;
        
        if (inicio == fim) {
            total = 24;
        } else if (inicio > fim) {
            fim = fim + 24;
            total = fim - inicio;
        } else {
            total = fim - inicio;
        }
        
        System.out.println("O JOGO DUROU " + total + " HORA(S)");
        
        sc.close();
        
    }
 
}